src/modules/hero/hero.component.ts

```diff
@@ -7,13 +7,41 @@ import { Component, Input } from '@angular/core';
 })
 export class StacheHeroComponent {
   @Input()
-  set backgroundImageUrl(imageUrl: string) {
-    this._backgroundImageUrl = imageUrl;
+  public backgroundImageUrl: string;
+
+  @Input()
+  set overlayOpacity(opacityCandidate: string) {
```

Comment:
This method is a lot better than it was, but I wouldn't be afraid to simplify it further.

Also, `toString` makes more immediate sense than `String()`...

```
@Input()
set overlayOpacity(value: string) {
  let sanitized = value.replace(/[^\d.-]/g, '');
  let interval = this.parseInterval(sanitized);
  this._overlayOpacity = interval.toString();
}
```